bokehjs/src/lib/models/grids/grid.ts

```diff
@@ -40,12 +38,18 @@ export class GridView extends GuideRendererView {
       const [sx0, sy0] = this.coordinates.map_to_screen(xs[i],   ys[i])
       const [sx1, sy1] = this.coordinates.map_to_screen(xs[i+1], ys[i+1])
 
-      if (this.visuals.band_fill.doit)
-        ctx.fillRect(sx0[0], sy0[0], sx1[1] - sx0[0], sy1[1] - sy0[0])
+      ctx.beginPath()
+      ctx.rect(sx0[0], sy0[0], sx1[1] - sx0[0], sy1[1] - sy0[0])
+
+      if (this.visuals.band_fill.doit) {
+        this.visuals.band_fill.set_value(ctx)
+        ctx.fill()
+      }
 
-      this.visuals.band_hatch.doit2(ctx, () => {
-        ctx.fillRect(sx0[0], sy0[0], sx1[1] - sx0[0], sy1[1] - sy0[0])
-      }, () => this.request_render())
```

Comment:
What causes this to no longer be necessary? It was there because hatch fills can be remote images that load asynchronously and previously they would not show up without explicitly scheduling a new render

Reply:
This will be handled the same way as images are loaded in `ImageURL` glyph, unifying all types of remote resource loading and it will fix readiness notification protocol, which this didn't implement.

Reply:
just making sure I understand, this is still TBD? I looked for something like what `ImageURL` does and did not see it